mozilla/fx-private-relay
# Add a plan feature comparison matrix

# New feature description

This adds a feature matrix comparing different plans. It's only visible if the `intro_pricing_countdown` flag is enabled and the pricing countdown has finished. It only shows the relevant plans and features if that plan's flag is enabled and it's available in the user's current country.

Sorry for this chonker. You might want to split the review up by sessions of a single commit at a time. (Edit: split some stuff out into https://github.com/mozilla/fx-private-relay/pull/2471.) Test coverage is also below the threshold, and I think it'd be good to add some tests for all the different flag situations anyway, so I'll add those tomorrow.

# Screenshot (if applicable)

Desktop, Premium not available:

![image](https://user-images.githubusercontent.com/4251/188687319-5baab7f1-d419-45a8-bf89-0bc77619bf06.png)

Mobile, Premium not available:

![image](https://user-images.githubusercontent.com/4251/188687373-42da4306-5cfa-4f96-810d-e33dd7c37558.png)

Desktop, only Premium available:

![image](https://user-images.githubusercontent.com/4251/188687461-cb50a70c-897c-4d23-9e02-75cad8322946.png)

Mobile, only Premium available:

![image](https://user-images.githubusercontent.com/4251/188687523-a73ef365-3fc1-48b0-b4b4-9dc6442745f7.png)

Desktop, bundle not available:

![image](https://user-images.githubusercontent.com/4251/188687592-7ac3bb1f-205a-4e84-bb31-51d0cedf6436.png)

Mobile, bundle not available:

![image](https://user-images.githubusercontent.com/4251/188687783-a462a312-bbd6-4cc9-8e44-2472a6314206.png)

Desktop, all available:

![image](https://user-images.githubusercontent.com/4251/188687877-5ad8c130-2f4a-4367-baa7-656a6ae7dfed.png)

Mobile, all available:

![image](https://user-images.githubusercontent.com/4251/188687908-82632b7d-19d3-4dc2-9447-fda1a5ce8617.png)


# How to test

Make sure the `intro_pricing_countdown` flag is enabled, and `INTRO_PRICING_END` is set to a date in the past (e.g. by adding `INTRO_PRICING_END = "2022-09-01T09:00:00.000-07:00"` to your `.env`). You should then be able to view the different plans on the landing page and `/premium`.

One open question I have: I can't get the purple VPN thing to look right on Chrome. Suggestions welcome:

![image](https://user-images.githubusercontent.com/4251/188688288-e771b923-408f-4586-939f-f05861a4d598.png)

# Checklist

- [ ] l10n changes have been submitted to the l10n repository, if any. - Not yet, think I'll do that together with Kaitlyn's changes?
- [ ] All acceptance criteria are met. - Almost, except that plan IDs and prices are not yet known.
